test/SlowTests/Issues/RavenDB_22107.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Threading.Tasks;
+using Raven.Tests.Core.Utils.Entities;
+using SlowTests.Sharding.Replication;
+using Tests.Infrastructure;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_22107 : ReplicationTestBase
+    {
+        public RavenDB_22107(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [RavenTheory(RavenTestCategory.Replication)]
+        [RavenData(DatabaseMode = RavenDatabaseMode.All)]
+        public async Task GetReplicationActiveConnectionsShouldNotThrow(Options options)
```

Comment:
The comment https://issues.hibernatingrhinos.com/issue/RavenDB-22107/ReplicationActiveConnectionsPreview-deserialization-is-broken#focus=Comments-67-1049081.0-0 says about "other friends" being affected as well. Can we add tests for them?

Reply:
I tried :) 
we have an issue with `IDictionary<TV, TK>` so I couldn't add a test for it, but I tested it manually.
I opened a dedicated issue for it (https://issues.hibernatingrhinos.com/issue/RavenDB-22177/Error-in-DynamicJsonValue.ConvertTK-TV-JsonDeserializationBase.ToDictionaryTK-TV-when-Deserializing-Non-Primitive-Types-as)

Reply:
hmmm I wonder if we should maybe use `KeyValuePair<>` here instead of a `Dictionary` 
how would you convert `Dictionary` with such key into json?

Reply:
yep, that's exactly the issue. I did find a potential solution, but it's quite convoluted, so we might want to avoid it:
- convert the `TK` object to a JSON string using `JsonConvert.SerializeObject`.
- instead of `(TK)Convert.ChangeType(propertyName, typeof(TK))`, call `JsonConvert.DeserializeObject` for `TK` at https://github.com/ravendb/ravendb/blob/v6.0/src/Sparrow/Json/JsonDeserializationBase.cs#L462.

while it seems to work for this particular case, there's a chance of unforeseen issues elsewhere. so, maybe `KeyValuePair<>` is a better choice